From 54c5d5e60ee069172f001d27f889edb4b78a6b10 Mon Sep 17 00:00:00 2001
From: Cameron Clark <cameron.clark@hey.com>
Date: Mon, 3 Nov 2025 12:44:32 +0000
Subject: [PATCH 4/7] fix: early return from invalid tsconfig for better errors

---
 internal/tsoptions/tsconfigparsing.go | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/internal/tsoptions/tsconfigparsing.go b/internal/tsoptions/tsconfigparsing.go
index 9668bb70b4..716a4e5d05 100644
--- a/internal/tsoptions/tsconfigparsing.go
+++ b/internal/tsoptions/tsconfigparsing.go
@@ -192,6 +192,17 @@ func parseOwnConfigOfJsonSourceFile(
 	var extendedConfigPath any
 	var rootCompilerOptions []*ast.PropertyName
 	var errors []*ast.Diagnostic
+
+	// If there are parsing errors, return only the first one
+	if len(sourceFile.Diagnostics()) > 0 {
+		return &parsedTsconfig{
+			raw:                &collections.OrderedMap[string, any]{},
+			options:            compilerOptions,
+			typeAcquisition:    typeAcquisition,
+			extendedConfigPath: extendedConfigPath,
+		}, []*ast.Diagnostic{sourceFile.Diagnostics()[0]}
+	}
+
 	onPropertySet := func(
 		keyText string,
 		value any,
-- 
2.50.1 (Apple Git-155)

